enhancements/insights/rapid-recommendations.md

```diff
@@ -0,0 +1,552 @@
+---
+title: rapid-recommendations
+authors:
+  - "@jholecek-rh"
+  - "@tremes"
+reviewers: 
+  - "@deads2k"
+approvers: 
+  - "@deads2k"
+api-approvers: 
+  - None
+creation-date: 2024-01-03
+last-updated: 2024-02-22
+tracking-link: 
+  - https://issues.redhat.com/browse/CCXDEV-12213
+  - https://issues.redhat.com/browse/CCXDEV-12285
+see-also:
+  - "/enhancements/insights/conditional-data-gathering.md"
+replaces:
+  - None
+superseded-by:
+  - None
+---
+
+# Insights Rapid Recommendations
+
+This enhancement proposal introduces a new approach to how the data collected 
+by the Insights operator can be defined remotely.
+
+
+## Summary
+
+The Insights Operator collects various data and resources from the OpenShift and Kubernetes APIs. 
+The definition of the collected data is mostly hardcoded in the operator's source code and largely 
+locked in the corresponding OCP version.
+
+This proposal introduces remote Insights Operator configuration for collected data. 
+The feature will allow Red Hat to control what data the Insights Operator gathers, 
+within hardcoded boundaries, independently of the cluster version. 
+
+
+## Motivation
+
+The Insights Operator collects various data and resources from the OpenShift and Kubernetes APIs. 
+The definition of the collected data is mostly hardcoded in the operator's source code 
+and largely locked in the corresponding OCP version.
+
+The data gathered by the Insights Operator can be divided into several groups based on their source 
+and format:
+
+* API resources - API resources are identified by the group, version, kind (GVK) and, optionally, 
+  namespace and name. 
+  They are represented as JSON or YAML objects.
+* Container logs - Container logs are identified by the namespace name, pod name, and messages of 
```

Comment:
Is it really pod _name_?  I'd have expected `(namespace, label, container name, messages of interest)`, like:

```console
$ oc -n openshift-machine-config-operator logs -l k8s-app=machine-config-daemon -c machine-config-daemon --tail -1 --since 30m | grep 'Starting to manage node:'
I0313 06:07:40.223963    1749 update.go:2341] Starting to manage node: build0-gstfj-ci-builds-worker-c-ln8qd
I0313 06:07:51.477713    1735 update.go:2341] Starting to manage node: build0-gstfj-ci-builds-worker-c-t86zg
I0313 05:53:01.826716    1736 update.go:2341] Starting to manage node: build0-gstfj-ci-longtests-worker-d-zszks
```

Reply:
Yes. It's not obvous from the text, but we plan to "scan" all the containers in the given Pod.